### PR TITLE
refactor!: mark `setData(..)` as payable

### PR DESCRIPTION
**Pull request by YamenMerhi**
## BREAKING CHANGE ⚠️ ⚠️ 

- mark `setData(..)` as payable
- Add tests for reverting when sending value

### CODE REVIEW

File: docs/ERC-725.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ The initial motivation came out of the need to create a smart contract account s
 
 This standard consists of two sub standards, a generic data key/value store (ERC725Y) and a generic execute function (ERC725X). Both of which in combination allow for a very flexible and long lasting account system. The account version of ERC725 is standardised under [LSP0 ERC725Account](https://github.com/lukso-network/LIPs/blob/master/LSPs/LSP-0-ERC725Account.md).
 
-These standareds (ERC725 X and Y) can also be used seperately to enhance NFTs and Token meta data or other types of smart contracts. ERC725X allows for a  generic exection through a smart contract, functioning as an account or actor.
+These standareds (ERC725 X and Y) can also be used seperately to enhance NFTs and Token meta data or other types of smart contracts. ERC725X allows for a generic exection through a smart contract, functioning as an account or actor.
```
Comment (YamenMerhi) [inline docs/ERC-725.md:24]:
```suggestion
These standards (ERC725 X and Y) can also be used separately to enhance NFTs and Token meta data or other types of smart contracts. ERC725X allows for a generic execution through a smart contract, functioning as an account or actor.
```

**Review by b00ste — APPROVED** [review]

File: implementations/contracts/ERC725YCore.sol [hunk 2]
```diff
@@ -56,19 +54,23 @@ abstract contract ERC725YCore is OwnableUnset, ERC165, IERC725Y {
...
-        public
-        virtual
-        override
-        onlyOwner
-    {
+    function setData(
+        bytes32[] memory dataKeys,
+        bytes[] memory dataValues
+    ) public payable virtual override onlyOwner {
+        if (msg.value != 0) revert ERC725Y_MsgValueDisallowed();
```
Comment (CJ42) [inline implementations/contracts/ERC725YCore.sol:72]: suggesting adding a comment.

```suggestion
        /// @dev do not allow to send value by default when setting data in ERC725Y
        if (msg.value != 0) revert ERC725Y_MsgValueDisallowed();
```

File: implementations/contracts/interfaces/IERC725Y.sol [hunk 3]
```diff
@@ -38,17 +38,27 @@ interface IERC725Y is IERC165 {
      * @param dataValue The value to set
      * SHOULD only be callable by the owner of the contract set via ERC173
      *
+     * The function was marked as payable to enable flexibility on child contracts
```
Comment (CJ42) [inline implementations/contracts/interfaces/IERC725Y.sol:41]:
```suggestion
     * The function is marked as payable to enable flexibility on child contracts
```

File: implementations/contracts/interfaces/IERC725Y.sol [hunk 4]
```diff
@@ -38,17 +38,27 @@ interface IERC725Y is IERC165 {
...
-    function setData(bytes32 dataKey, bytes memory dataValue) external;
+    function setData(bytes32 dataKey, bytes memory dataValue) external payable;
 
     /**
      * @param dataKeys The array of data keys for values to set
      * @param dataValues The array of values to set
      * @dev Sets array of data for multiple given `dataKeys`
      * SHOULD only be callable by the owner of the contract set via ERC173
      *
+     * The function was marked as payable to enable flexibility on child contracts
```
Comment (CJ42) [inline implementations/contracts/interfaces/IERC725Y.sol:56]:
```suggestion
     * The function is marked as payable to enable flexibility on child contracts
```

**Review by CJ42 — APPROVED** [review]

**Pull request merged**